### ISSUE TITLE
Test rewrite rules with `inspection-testing`?

### ISSUE THREAD

**Issue by michaelpj** [snowleopard/alga#129]
Just a drive-by thought since I noticed that you have some rewrite rules now. (inspection-testing)[https://github.com/nomeata/inspection-testing] is a nice library for testing that these are actually firing as you intend them to, without relying on inspecting benchmarks.

**Comment by snowleopard** [comment]
Good idea! @nobrakal and I wanted to test rewrite rules before but didn't know about this library then.

@nobrakal Have a look at this approach, it might help you a lot with your on-going work on rewrite rules.

There is a nice Haskell eXchange talk about inspection testing:

https://skillsmatter.com/skillscasts/12590-a-promise-checked-is-a-promise-kept-inspection-testing

**Comment by nobrakal** [comment]
This seems geat :D 

Indeed it can really help me testing rule firings, and as @michaelpj  pointed it out, it seems that this library can simply test that all is working good in alga's test-suite.

I definitely need to dig on this, thanks for the link !

**Comment by snowleopard** [comment]
@nobrakal After merging your draft implementation of inspection testing, my `ghcid` setup is no longer working: the RewriteRules testsuite reports errors, I guess because `ghcid` compiles with no optimisation... Do you know if it's possible to switch off inspection testing with a flag somehow?
